cmd/kubeadm/app/util/pkiutil/pki_helpers.go

```diff
@@ -81,13 +81,13 @@ func NewCertificateAuthority(config *CertConfig) (*x509.Certificate, crypto.Sign
 }
 
 // NewIntermediateCertificateAuthority creates new certificate and private key for an intermediate certificate authority
-func NewIntermediateCertificateAuthority(parentCert *x509.Certificate, parentKey crypto.Signer, config *CertConfig) (*x509.Certificate, crypto.Signer, error) {
+func NewIntermediateCertificateAuthority(parentCert *x509.Certificate, parentKey crypto.Signer, config *CertConfig, validityPeriod time.Duration) (*x509.Certificate, crypto.Signer, error) {
```

Comment:
in the future when pki_helpers is part of a kubeadm library we won't be able to so easily change the signature / arguments of these exported functions.

i think instead of passing this as an additional argument, we should use CertConfig (defined in pki_helpers)
and have a field `NotAfter` of type `*time.Time` in there.

- keep NotBefore hardcoded for the time being.
- if the user does not pass a value for "NotAfter" (NotAfter == nil), default to `time.Now().Add(kubeadmconstants.CertificateValidity).UTC()` in `NewSignedCert()`.

this should work.

Reply:
make sense!
thanks @neolit123